### PR TITLE
Update image-card to use govuk-frontend

### PR DESCRIPTION
**Pull request by alex-ju**
## What
Update `image-card.scss` to use govuk-frontend mixins and variables and refactor code to be mobile first
Fix source path for [missing image](https://govuk-publishing-components.herokuapp.com/component-guide/image_card/large_version/preview).

## Why
To help [removing the dependency on the deprecated govuk_frontend_toolkit package](https://trello.com/c/ajPzDAOX).

## Visual Changes
No visual changes. Attached desktop and mobile diff (left is master, right is current branch)

### Desktop
image-card-desktop

### Mobile
image-card-mobile

### CODE REVIEW

**Comment by andysellick** [comment]
@alex-ju can you make the review app work?

File: app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss [hunk 1]
```diff
@@ -69,26 +72,26 @@
 
 .gem-c-image-card__context,
 .gem-c-image-card__metadata {
-  @include core-14;
-  margin: 0 0 $gutter-one-quarter;
-  color: $grey-1;
+  @include govuk-font(14);
+  margin: 0 0 (govuk-spacing(3) / 2);
```
Comment (andysellick) [inline app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss]: At some point (not now) we should have a discussion about the weirder spacing required on GOV.UK, and whether it either needs to be brought in line with the spacing scale or just done as a number - things like this are a bit hard to read sometimes.

File: app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss [hunk 2]
```diff
@@ -97,58 +100,58 @@
...
+    @include govuk-grid-column(two-thirds, $class: false, $at: tablet);
 
-  .gem-c-image-card__text-wrapper {
-    overflow: hidden;
-  }
+    margin-bottom: govuk-spacing(2);
+    float: none;
+    width: auto;
+    max-width: 100%;
+    padding: 0;
```
Comment (andysellick) [inline app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss]: `padding: 0` is being declared both here and on line 140 for I think the same element? Is this necessary?

File: app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss [hunk 3]
```diff
@@ -1,25 +1,27 @@
 .gem-c-image-card {
   // if this extends grid-row a margin-bottom can't
   // be applied as the extend overrides it
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss]: Nope. Removed it.

File: app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
 
   + .gem-c-image-card__text-wrapper {
-    @include media(mobile) {
+    @include govuk-media-query($until: tablet) {
       padding-left: 0;
     }
   }
+
+  @include govuk-grid-column(one-third, $class: false, $at: mobile);
+  @include govuk-grid-column(full, $class: false, $at: tablet);
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss]: Moved them upwards. Good point about these, we may want to document it somewhere.

File: app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss (hunk 1 above)
Comment (alex-ju) [reply app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss]: Agreed

File: app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss (hunk 2 above)
Comment (alex-ju) [reply app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss]: One is on `gem-c-image-card__image-wrapper` the other on `gem-c-image-card__text-wrapper`, two different elements I'm afraid.

**Comment by alex-ju** [comment]
Thanks for the review @andysellick I've updated based on your comments.

**Pull request merged**